src/Chainweb/Pact/Backend/ChainwebPactDb.hs

```diff
@@ -352,13 +393,21 @@ doKeys d = do
     return allKeys
 
   where
+    blockLimitStmt = maybe "" (const " WHERE txid < ?;") mlim
+    blockLimitParam = maybe [] (\(TxId txid) -> [SInt (fromIntegral txid)]) (snd <$> mlim)
     getDbKeys = do
-        m <- runMaybeT $ checkDbTableExists $ Utf8 tnS
+        m <- runMaybeT $ checkDbTablePendingCreation $ Utf8 tnS
         case m of
             Nothing -> return mempty
             Just () -> do
+                case mlim of
+                    Nothing -> return ()
+                    Just (bh, _) -> do
+                        exists <- tableExists tn bh
+                        unless exists $ -- TODO: try to throw a DbError more cleanly. comment this.
+                            void $ callDb "doKeys" $ \db -> qry db "garbage query" [] []
```

Comment:
Pull this out since it happens in a few places; `failIfTableDoesNotExist` or something like that